### ISSUE TITLE
Switch to new name `oneDNN` instead of `MKLDNN`

### ISSUE THREAD

**Issue by yanbing-j** [pytorch/pytorch#132368]
MKLDNN was renamed to oneDNN in 2020. And it's available on Arm of course. We can gradually switch to the new name oneDNN.

We plan to rename all ‘mkldnn’ to ‘onednn’, meanwhile, keep user facing APIs with ‘mkldnn’ and create user facing APIs with ‘onednn’. At one point in the near future, we will deprecate user facing APIs with ‘mkldnn’.

**Comment by malfet** [comment]
Yes, let's do this, though if we are about to change any APIs, we need to be careful about breaking backward compatibility (I suggest keeping an old name as an alias for a while)

**Comment by malfet** [comment]
Also, perhaps it's a good time to get rid of `ideep` wrapper (or may be just embed entire functionality into PyTorch and call oneDNN directly)

**Comment by vpirogov** [comment]
> Also, perhaps it's a good time to get rid of `ideep` wrapper (or may be just embed entire functionality into PyTorch and call oneDNN directly)

That was my dream for at least last three years.

**Comment by jgong5** [comment]
> Also, perhaps it's a good time to get rid of `ideep` wrapper (or may be just embed entire functionality into PyTorch and call oneDNN directly)

I guess embedding ideep into PyTorch (just code movement) could be a reasonable first step in the PyTorch 2.5 timeline. cc @Guobing-Chen @yanbing-j

**Comment by yanbing-j** [comment]
@malfet We could switch name to `oneDNN` first.

After it is merged, we can start to move ideep code into PyTorch. To be noticed, there is large amount of ideep codes. We also need time to clean the code when do the moving. And not sure if encounters issues, e.g, build and performance. It has risk to target PyTorch 2.5. cc @Guobing-Chen @jgong5
